frontend/awx/access/credentials/hooks/useCredentialsFilters.tsx

```diff
@@ -1,25 +1,18 @@
-import { useMemo } from 'react';
-import { IToolbarFilter } from '../../../../../framework';
+import { useDynamicToolbarFilters } from '../../../common/useDynamicFilters';
 import {
   useCreatedByToolbarFilter,
-  useDescriptionToolbarFilter,
   useModifiedByToolbarFilter,
-  useNameToolbarFilter,
 } from '../../../common/awx-toolbar-filters';
 
 export function useCredentialsFilters() {
-  const nameToolbarFilter = useNameToolbarFilter();
-  const descriptionToolbarFilter = useDescriptionToolbarFilter();
   const createdByToolbarFilter = useCreatedByToolbarFilter();
   const modifiedByToolbarFilter = useModifiedByToolbarFilter();
-  const toolbarFilters = useMemo<IToolbarFilter[]>(
-    () => [
-      nameToolbarFilter,
-      descriptionToolbarFilter,
-      createdByToolbarFilter,
-      modifiedByToolbarFilter,
-    ],
-    [nameToolbarFilter, descriptionToolbarFilter, createdByToolbarFilter, modifiedByToolbarFilter]
-  );
-  return toolbarFilters;
+  const toolBarFilters = useDynamicToolbarFilters({
+    optionsPath: 'credentials',
+    preSortedKeys: ['name'],
```

Comment:
I would put credential type towards the top of the filter list since it is prefilled. I would do the same with created by and modified by since those were additionally added. So the order would be name, credential type, created by, modified by- just my two cents

Reply:
Done! Also put ID in the preFilledValueKeys.